dart/constraint/ConstrainedGroup.h

```diff
@@ -107,7 +108,7 @@ class ConstrainedGroup
   std::vector<ConstraintBase*> mConstraints;
 
   ///
-  dynamics::Skeleton* mRootSkeleton;
+  std::shared_ptr<dynamics::Skeleton> mRootSkeleton;
```

Comment:
I agree that including unnecessary header increases the compilation times.

Then we might create many of `[class]Ptr.h` files. How about creating `DynamicsTypes.h` and put all the smart pointer types into the file like:
```cpp
// DynamicsTypes.h

#include <memory>

namespace dart {
namespace dynamics {

class Skeleton;
typedef std::shared_ptr<Skeleton> SkeletonPtr;

class Shape;
typedef std::shared_ptr<Shape> ShapePtr;

// and so on

} // namespace dynamics
} // namespace dart
```